### PR TITLE
HSEARCH-3429 Add a lambda-based syntax to the projection DSL

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3429

### CODE REVIEW

File: documentation/src/main/asciidoc/mapper-pojo.asciidoc [hunk 1]
```diff
@@ -22,6 +22,139 @@ TODO The getting started section has a link pointing here and expects the sectio
...
+Full-text fields must be assigned an `analyzer`, referenced by its name.
+See <<analysis>> for more details about analyzers and full-text analysis.
++
+IMPORTANT: Full-text fields cannot be sorted on.
+If you need to sort on the value of a property,
+it is recommended to use `@KeywordField`, with a normalizer if necessary (see below).
+Note that multiple fields can be added to the same
+`@KeywordField`::
+A text field whose value is considered as a single keyword.
+Only works for String fields.
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapper-pojo.asciidoc]: Either `String` or string

File: documentation/src/main/asciidoc/mapper-pojo.asciidoc [hunk 2]
```diff
@@ -22,6 +22,139 @@ TODO The getting started section has a link pointing here and expects the sectio
...
++
+Keyword fields allow subtle matches, similarly to full-text fields,
+with the limitation that keyword fields only contain one word.
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapper-pojo.asciidoc]: I would say one token, not one word.

File: documentation/src/main/asciidoc/mapper-pojo.asciidoc [hunk 3]
```diff
@@ -69,6 +202,32 @@ The corresponding annotation is `org.hibernate.search.integrationtest.showcase.l
...
 
+=== Value bridges
+
+include::todo-placeholder.asciidoc[]
+
+[[mapper-pojo-bridge-valuebridge-builtin]]
+===== Built-in value bridges
+
+IMPORTANT: There are few supported types at the moment.
+This will be solved before the 6.0.0.Final release, see https://hibernate.atlassian.net/browse/HSEARCH-3047[HSEARCH-3047].
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapper-pojo.asciidoc]: Maybe a proper Asciidoc block would be nice?

File: documentation/src/main/asciidoc/mapper-pojo.asciidoc [hunk 4]
```diff
@@ -69,6 +202,32 @@ The corresponding annotation is `org.hibernate.search.integrationtest.showcase.l
 * `org.hibernate.search.integrationtest.showcase.library.bridge.AccountBorrowalSummaryBridge` implements `TypeBridge`.
 The corresponding annotation is `org.hibernate.search.integrationtest.showcase.library.bridge.annotation.AccountBorrowalSummaryBridge`.
 
+=== Value bridges
+
+include::todo-placeholder.asciidoc[]
+
+[[mapper-pojo-bridge-valuebridge-builtin]]
+===== Built-in value bridges
```
Comment (gsmet) [inline documentation/src/main/asciidoc/mapper-pojo.asciidoc]: I think you have one too many =

File: documentation/src/main/asciidoc/mapper-pojo.asciidoc (hunk 3 above)
Comment (yrodiere) [reply documentation/src/main/asciidoc/mapper-pojo.asciidoc]: This is the short syntax. This is rendered as a proper Asciidoc admonition.

**Comment by yrodiere** [comment]
Fixed the problems you mentioned and merged. Thanks!

**Pull request merged**